File: 392/main.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        """
        approach
        2 pointers
        if a == b
        advance both
        """
        a, b = 0, 0
        while True:
            if a == len(s):
                return True
            elif b == len(t):
                return False
            ch_a, ch_b = s[a], t[b]
            if ch_a == ch_b:
                a += 1
                b += 1
            else:
                b += 1

File: 392/main.ts
function isSubsequence(s: string, t: string): boolean {
  let subsequenceIndex = 0;
  for (let i = 0; i < t.length; i++) {
    if (subsequenceIndex === s.length) return true;
    if (t[i] === s[subsequenceIndex]) subsequenceIndex++;
  }
  return subsequenceIndex === s.length;
}
